File: OOP C++/lab2/Over.cpp
#include "Over.h"
#include "FactoryInitializer.h"

void Over::Execute(Operands& operands, Tokens& tokens, std::string& output, Reader& reader) {
    int operand1 = operands.GetAndPop();
    int operand2 = operands.Top();

    operands.Push(operand1);
    operands.Push(operand2);

}

namespace {
    FactoryInitializer<Over> Registration("over");
}

File: OOP Java/lab1/src/main/java/org/example/commands/Over.java
package org.example.commands;

import org.example.utility.Operands;

import java.util.StringTokenizer;

public class Over implements Command{
    @Override
    public void Execute(StringTokenizer tokens, Operands operands) {
        int num1 = operands.GetAndPop();
        int num2 = operands.Top();
        operands.Push(num1);
        operands.Push(num2);
    }

    @Override
    public void Check(StringTokenizer tokens, Operands operands) {

    }

}
